### ISSUE TITLE
Long branch names cause auth failure for workload identity federation

### ISSUE THREAD

**Issue by kylekurz** [google-github-actions/auth#124]
### TL;DR

Using a long branch name will result in an invalid subject line in the request, resulting in the following error:

```
google-github-actions/auth failed with: failed to generate Google Cloud federated token for projects/<project>/locations/global/workloadIdentityPools/github-identity-pool/providers/github-provider: {"error":"invalid_request","error_description":"The size of mapped attribute google.subject exceeds the 127 bytes limit. Either modify your attribute mapping or the incoming assertion to produce a mapped attribute that is less than 127 bytes."}
```

*NOTE:* I edited this to include a simpler log, action.yaml and use the latest version of the plugin.

### Expected behavior

Should be able to use any valid GitHub ref to trigger a WIF authentication.

### Observed behavior

WIF request fails and the job cannot proceed.

### Action YAML

```yaml
name: GitHub Bug Report
on: [push]
jobs:
  deploy:
    permissions:
      contents: "read"
      id-token: "write"
    runs-on: ubuntu-20.04
    steps:
      - name: Check out repository
        uses: actions/checkout@v2
        with:
          fetch-depth: 0
      - id: "auth"
        name: "Authenticate to Google Cloud"
        uses: "google-github-actions/auth@v0.5.0"
        with:
          token_format: "access_token"
          workload_identity_provider: "projects/<redacted>/locations/global/workloadIdentityPools/github-identity-pool/providers/github-provider"
          service_account: "<redacted>"
          create_credentials_file: true
```

### Log output

```text
2022-02-02T17:54:18.2078222Z Waiting for a runner to pick up this job...
2022-02-02T17:54:18.7663541Z Job is waiting for a hosted runner to come online.
2022-02-02T17:54:21.5957954Z Job is about to start running on the hosted runner: GitHub Actions 5 (hosted)
2022-02-02T17:54:25.0873159Z Current runner version: '2.287.1'
2022-02-02T17:54:25.0900688Z ##[group]Operating System
2022-02-02T17:54:25.0901172Z Ubuntu
2022-02-02T17:54:25.0901445Z 20.04.3
2022-02-02T17:54:25.0901649Z LTS
2022-02-02T17:54:25.0901987Z ##[endgroup]
2022-02-02T17:54:25.0902248Z ##[group]Virtual Environment
2022-02-02T17:54:25.0902601Z Environment: ubuntu-20.04
2022-02-02T17:54:25.0902910Z Version: 20220123.1
2022-02-02T17:54:25.0903350Z Included Software: https://github.com/actions/virtual-environments/blob/ubuntu20/20220123.1/images/linux/Ubuntu2004-Readme.md
2022-02-02T17:54:25.0903995Z Image Release: https://github.com/actions/virtual-environments/releases/tag/ubuntu20%2F20220123.1
2022-02-02T17:54:25.0904367Z ##[endgroup]
2022-02-02T17:54:25.0904702Z ##[group]Virtual Environment Provisioner
2022-02-02T17:54:25.0905032Z 1.0.0.0-main-20220128-1
2022-02-02T17:54:25.0905274Z ##[endgroup]
2022-02-02T17:54:25.0905851Z ##[group]GITHUB_TOKEN Permissions
2022-02-02T17:54:25.0906337Z Contents: read
2022-02-02T17:54:25.0906759Z Metadata: read
2022-02-02T17:54:25.0907050Z ##[endgroup]
2022-02-02T17:54:25.0910740Z Secret source: Actions
2022-02-02T17:54:25.0911161Z Prepare workflow directory
2022-02-02T17:54:25.1782465Z Prepare all required actions
2022-02-02T17:54:25.1970927Z Getting action download info
2022-02-02T17:54:25.4323218Z Download action repository 'actions/checkout@v2' (SHA:ec3a7ce113134d7a93b817d10a8272cb61118579)
2022-02-02T17:54:26.4517066Z Download action repository 'google-github-actions/auth@v0.5.0' (SHA:7a360a247cce1245c1d13bc63d8b2ac927d201c8)
2022-02-02T17:54:27.3492832Z ##[group]Run actions/checkout@v2
2022-02-02T17:54:27.3493132Z with:
2022-02-02T17:54:27.3493489Z   fetch-depth: 0
2022-02-02T17:54:27.3493742Z   repository: <redacted>
2022-02-02T17:54:27.3494292Z   token: ***
2022-02-02T17:54:27.3494513Z   ssh-strict: true
2022-02-02T17:54:27.3494800Z   persist-credentials: true
2022-02-02T17:54:27.3495125Z   clean: true
2022-02-02T17:54:27.3495324Z   lfs: false
2022-02-02T17:54:27.3495569Z   submodules: false
2022-02-02T17:54:27.3495797Z ##[endgroup]
2022-02-02T17:54:27.6004388Z Syncing repository: <redacted>
2022-02-02T17:54:27.6006214Z ##[group]Getting Git version info
2022-02-02T17:54:27.6006874Z Working directory is '/home/runner/work/<redacted>'
2022-02-02T17:54:27.6007493Z [command]/usr/bin/git version
2022-02-02T17:54:27.6098663Z git version 2.34.1
2022-02-02T17:54:27.6119945Z ##[endgroup]
2022-02-02T17:54:27.6127428Z Deleting the contents of '/home/runner/work/<redacted>'
2022-02-02T17:54:27.6133584Z ##[group]Initializing the repository
2022-02-02T17:54:27.6138821Z [command]/usr/bin/git init /home/runner/work/<redacted>
2022-02-02T17:54:27.6215448Z Initialized empty Git repository in /home/runner/work/<redacted>/.git/
2022-02-02T17:54:27.6227577Z [command]/usr/bin/git remote add origin https://github.com/<redacted>
2022-02-02T17:54:27.6268913Z ##[endgroup]
2022-02-02T17:54:27.6269790Z ##[group]Disabling automatic garbage collection
2022-02-02T17:54:27.6272699Z [command]/usr/bin/git config --local gc.auto 0
2022-02-02T17:54:27.6305351Z ##[endgroup]
2022-02-02T17:54:27.6324625Z ##[group]Setting up auth
2022-02-02T17:54:27.6326311Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2022-02-02T17:54:27.6351475Z [command]/usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :
2022-02-02T17:54:27.6701367Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2022-02-02T17:54:27.6763482Z [command]/usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :
2022-02-02T17:54:27.6984237Z [command]/usr/bin/git config --local http.https://github.com/.extraheader AUTHORIZATION: basic ***
2022-02-02T17:54:27.7029115Z ##[endgroup]
2022-02-02T17:54:27.7030556Z ##[group]Fetching the repository
2022-02-02T17:54:27.7038145Z [command]/usr/bin/git -c protocol.version=2 fetch --prune --progress --no-recurse-submodules origin +refs/heads/*:refs/remotes/origin/* +refs/tags/*:refs/tags/*
2022-02-02T17:54:28.4684959Z remote: Enumerating objects: 4616, done.        
2022-02-02T17:54:28.4685597Z remote: Counting objects:   0% (1/643)        
<simplified>
2022-02-02T17:54:32.6721590Z Resolving deltas: 100% (3018/3018), done.
2022-02-02T17:54:32.6961896Z From https://github.com/<redacted>
2022-02-02T17:54:32.6965024Z  * [new branch]      oleksii/2746-add-a-button-in-madmin-that-is-a-shortcut-to-the-given-merchant-in-the-sales-dashboard -> origin/oleksii/2746-add-a-button-in-madmin-that-is-a-shortcut-to-the-given-merchant-in-the-sales-dashboard
2022-02-02T17:54:32.6993315Z [command]/usr/bin/git branch --list --remote origin/oleksii/2746-add-a-button-in-madmin-that-is-a-shortcut-to-the-given-merchant-in-the-sales-dashboard
2022-02-02T17:54:32.7021679Z   origin/oleksii/2746-add-a-button-in-madmin-that-is-a-shortcut-to-the-given-merchant-in-the-sales-dashboard
2022-02-02T17:54:32.7031406Z [command]/usr/bin/git rev-parse refs/remotes/origin/oleksii/2746-add-a-button-in-madmin-that-is-a-shortcut-to-the-given-merchant-in-the-sales-dashboard
2022-02-02T17:54:32.7056803Z c3ffd0ff38303a6295ebd6e14d880fdea1d89dc3
2022-02-02T17:54:32.7062236Z ##[endgroup]
2022-02-02T17:54:32.7062870Z ##[group]Determining the checkout info
2022-02-02T17:54:32.7066351Z ##[endgroup]
2022-02-02T17:54:32.7067959Z ##[group]Checking out the ref
2022-02-02T17:54:32.7074346Z [command]/usr/bin/git checkout --progress --force -B oleksii/2746-add-a-button-in-madmin-that-is-a-shortcut-to-the-given-merchant-in-the-sales-dashboard refs/remotes/origin/oleksii/2746-add-a-button-in-madmin-that-is-a-shortcut-to-the-given-merchant-in-the-sales-dashboard
2022-02-02T17:54:32.7305267Z Switched to a new branch 'oleksii/2746-add-a-button-in-madmin-that-is-a-shortcut-to-the-given-merchant-in-the-sales-dashboard'
2022-02-02T17:54:32.7306709Z Branch 'oleksii/2746-add-a-button-in-madmin-that-is-a-shortcut-to-the-given-merchant-in-the-sales-dashboard' set up to track remote branch 'oleksii/2746-add-a-button-in-madmin-that-is-a-shortcut-to-the-given-merchant-in-the-sales-dashboard' from 'origin'.
2022-02-02T17:54:32.7324554Z ##[endgroup]
2022-02-02T17:54:32.7352521Z [command]/usr/bin/git log -1 --format='%H'
2022-02-02T17:54:32.7380757Z 'c3ffd0ff38303a6295ebd6e14d880fdea1d89dc3'
2022-02-02T17:54:32.7581584Z ##[group]Run google-github-actions/auth@v0.5.0
2022-02-02T17:54:32.7581916Z with:
2022-02-02T17:54:32.7582104Z   token_format: access_token
2022-02-02T17:54:32.7582485Z   workload_identity_provider: projects/<redacted>/locations/global/workloadIdentityPools/github-identity-pool/providers/github-provider
2022-02-02T17:54:32.7582941Z   service_account: <redacted>
2022-02-02T17:54:32.7583257Z   create_credentials_file: true
2022-02-02T17:54:32.7583466Z   cleanup_credentials: true
2022-02-02T17:54:32.7583686Z   access_token_lifetime: 3600s
2022-02-02T17:54:32.7583982Z   access_token_scopes: https://www.googleapis.com/auth/cloud-platform
2022-02-02T17:54:32.7584258Z   id_token_include_email: false
2022-02-02T17:54:32.7584454Z ##[endgroup]
2022-02-02T17:54:34.6517813Z Created credentials file at "/home/runner/work/<redacted>"
2022-02-02T17:54:34.6518355Z 
2022-02-02T17:54:34.6545833Z 
2022-02-02T17:54:34.7859492Z ##[error]google-github-actions/auth failed with: failed to generate Google Cloud federated token for projects/<redacted>/locations/global/workloadIdentityPools/github-identity-pool/providers/github-provider: {"error":"invalid_request","error_description":"The size of mapped attribute google.subject exceeds the 127 bytes limit. Either modify your attribute mapping or the incoming assertion to produce a mapped attribute that is less than 127 bytes."}
2022-02-02T17:54:34.7929396Z Post job cleanup.
2022-02-02T17:54:34.8544512Z Removed exported credentials at "/home/runner/work/<redacted>".
2022-02-02T17:54:34.8635780Z Post job cleanup.
2022-02-02T17:54:34.9804895Z [command]/usr/bin/git version
2022-02-02T17:54:34.9847194Z git version 2.34.1
2022-02-02T17:54:34.9879976Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2022-02-02T17:54:34.9914118Z [command]/usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :
2022-02-02T17:54:35.0139163Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2022-02-02T17:54:35.0164965Z http.https://github.com/.extraheader
2022-02-02T17:54:35.0175788Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2022-02-02T17:54:35.0208839Z [command]/usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :
2022-02-02T17:54:35.0592499Z Cleaning up orphan processes
```

### Additional information

~~I can upgrade to `v0.5.0` and try again, but the changelog did not appear to have anything that would adjust this behavior. Please let me know if you'd like me to do that and report back.~~

**Comment by sethvargo** [comment]
Hi @kylekurz 

This is a limit imposed by Google Cloud IAM. We have no control over this value. Please [file feedback with the Google Cloud IAM team](https://cloud.google.com/iam/docs/getting-support). The only mitigation is to use shorter repo names or shorter branch names. Sorry!

**Comment by kylekurz** [comment]
@sethvargo I guess my question is more... why use the branch name instead of the hash ref? That would still be unique and identifiable, but would also remain well under the 127 character limit. Or cut it off at 100 characters, being _mostly_ unique. 

At the very least, should that be a documented limitation on this action? I don't see anything in the README currently, but It was pretty surprising to run into.

**Comment by sethvargo** [comment]
Hi @kylekurz - I opened https://github.com/google-github-actions/auth/pull/127 to add more documentation and troubleshooting steps. This is mentioned there.

I think there might be a misunderstanding. We have no control over the claims in the token. The claims and their values are entirely [managed by GitHub](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#understanding-the-oidc-token), which creates, signs, and injects the token into the environment. They define the `sub`, not us.

**Comment by kylekurz** [comment]
@sethvargo ahh, makes sense. Thanks for taking the time to explain to me. I'll document it for our team as well.

**Comment by neelance** [comment]
I managed to fix this by using https://docs.github.com/en/rest/actions/oidc to set `include_claim_keys` to only `repo`. Important: You need to set it on the repository itself. Setting it on the organisation only makes new repositories inherit the setting.
